### ISSUE TITLE
Service Accounts

### ISSUE THREAD

**Issue by johnrutherford** [IdentityServer/IdentityServer4#1353]
We need to allow our users to create "service accounts" to call our APIs. These accounts would be used by service code with no user interaction and should have long lived access (probably using refresh tokens), but be revokeable. 

We also want the access tokens used by the service accounts to have the identity of the user that created the account. 

Any advice on how to accomplish this with IdentityServer4?

**Comment by leastprivilege** [comment]
In the OAuth world - service accounts the way you describe them are clients using the client credentials grant.

**Comment by johnrutherford** [comment]
Thank you for the response. So I can create an interface to allow each user to create their own client. Then they would use the client credentials grant to get an access token/refresh token,, etc. 

How would make it so the access token would include the user claims (e.g., sub, role, etc.)?

**Comment by leastprivilege** [comment]
I am not following - clients are not users. Please check the specs / concepts first.

**Comment by johnrutherford** [comment]
I understand the difference between clients and users. Let me rephrase my original question.

I need to allow our users to create some sort of "API credentials" that they can use to obtain access tokens to call our APIs. The access tokens should contain information about the user that created the API credentials. The flow needs to be non-interactive since these credentials will be used by background services.

**Comment by johnrutherford** [comment]
Ok, maybe I'm going about this wrong. 

I think I just need to allow users to create a client and then use the client credentials grant to obtain tokens. 

We currently have users calling our APIs and we do authorization checks against the principal. So for simplicity, I wanted calls from users's "service accounts" to look like the user. But I guess it makes more sense to allow calls from clients and figure out how to authorize clients as well as users.

**Comment by brockallen** [comment]
All set on this issue? Can we close?

**Comment by johnrutherford** [comment]
Yes
